.github/workflows/build_recipes.yaml

```diff
@@ -2,18 +2,96 @@ name: build recipes
 on:
   push:
     branches:
-      - 'main'
+      - "main"
     paths:
-      - 'recipes/**'
+      - "recipes/**"
   pull_request:
     paths:
-      - 'recipes/**'
-
+      - "recipes/**"
 
 jobs:
+  lint_recipes:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ github.sha }}
+
+      - name: Fetch main branch
+        run: |
+          git fetch origin main
+
+      - name: Install micromamba
+        uses: mamba-org/setup-micromamba@v1
+        with:
+          micromamba-version: "1.5.8-0"
+          environment-file: ci_env.yml
+          environment-name: ci
+          init-shell: bash
+          cache-environment: false
+          post-cleanup: "all"
+
+      - name: Install yq
+        run: sudo apt-get update && sudo apt-get install -y yq
+
+      - name: Lint recipes for license fields
+        shell: bash -el {0}
+        run: |
+          set -e
+
+          if [ "${{ github.event_name }}" = "pull_request" ]; then
+            BASE_REF="origin/main"
+            HEAD_REF="HEAD"
+          else
+            BASE_REF="HEAD~1"
+            HEAD_REF="HEAD"
+          fi
+
+          echo "Comparing changes between $BASE_REF and $HEAD_REF"
+
+          changed_recipes_json=$(python -m emci.find_recipes_with_changes $BASE_REF $HEAD_REF)
```

Comment:
I think i am failing to get the usage right here.

Reply:
`find_recipes_with_changes` that is not indeded to be called from the cli

Reply:
Thanks for clarifying Thorsten, Is there any other way around you may have in mind to get the `changed_recipes`?